internal/collector/config_test.go

```diff
@@ -61,3 +63,134 @@ service:
 `)
 	})
 }
+
+// TODO: write this test after rebasing on new retention API changes.
+func TestGenerateLogrotateConfig(t *testing.T) {
+	for _, tt := range []struct {
+		logFilePath      string
+		retentionPeriod  string
+		postrotateScript string
+		result           string
+	}{
+		{
+			logFilePath:      "/this/is/a/file.path",
+			retentionPeriod:  "12h",
+			postrotateScript: "echo 'Hello, World'",
+			result: `/this/is/a/file.path {
+      rotate 12
+      missingok
+      sharedscripts
+      notifempty
+      nocompress
+      hourly
+      postrotate
+            echo 'Hello, World'
+      endscript
+}
+`,
+		},
+		{
+			logFilePath:      "/tmp/test.log",
+			retentionPeriod:  "5 days",
+			postrotateScript: "",
+			result: `/tmp/test.log {
+      rotate 5
+      missingok
+      sharedscripts
+      notifempty
+      nocompress
+      daily
+      postrotate
+            
+      endscript
+}
+`,
+		},
+		{
+			logFilePath:      "/tmp/test.log",
+			retentionPeriod:  "5wk",
+			postrotateScript: "pkill -HUP --exact pgbouncer",
+			result: `/tmp/test.log {
+      rotate 35
+      missingok
+      sharedscripts
+      notifempty
+      nocompress
+      daily
+      postrotate
+            pkill -HUP --exact pgbouncer
+      endscript
+}
+`,
+		},
+	} {
+		t.Run(tt.retentionPeriod, func(t *testing.T) {
+			duration, err := v1beta1.NewDuration(tt.retentionPeriod)
+			assert.NilError(t, err)
+			result := generateLogrotateConfig(tt.logFilePath, duration, tt.postrotateScript)
+			assert.Equal(t, tt.result, result)
+		})
+	}
+}
+
+func TestParseDurationForLogrotate(t *testing.T) {
+	for _, tt := range []struct {
+		retentionPeriod string
+		number          int
+		interval        string
+	}{
+		{
+			retentionPeriod: "12h",
+			number:          12,
+			interval:        "hourly",
+		},
+		{
+			retentionPeriod: "24hr",
+			number:          1,
+			interval:        "daily",
+		},
+		{
+			retentionPeriod: "35hour",
+			number:          1,
+			interval:        "daily",
+		},
+		{
+			retentionPeriod: "36 hours",
+			number:          2,
+			interval:        "daily",
+		},
```

Comment:
👍🏻 These tests!

🤔 Does logrotate "1 daily" mean (1) current file + (1) one day saved? If so, the contents of the two files combined is between 24 and 48 hours, right?
- just before rotation = 24 hours + 24 hours
- just after rotation = 0 hours + 24 hours

🔧 If that's how it works, I think I want both these cases to be `2, "daily"` so that right after rotation, we retain (more than) 35 hours. (Swap `math.Round` for `math.Ceil`? I dunno!)

Reply:
Ahh yes, this makes sense. I'll change it.